src/cpp/server/server_context.cc

```diff
@@ -157,14 +157,15 @@ class BackendMetricState : public experimental::CallMetricRecorder,
   }
 
  private:
+  // Once returned metrics recorded with CallMetricRecorder APIs are cleared.
   grpc_core::BackendMetricData GetBackendMetricData() override
       ABSL_NO_THREAD_SAFETY_ANALYSIS {
-    grpc_core::BackendMetricData data;
     // Merge metrics from the ServerMetricRecorder first since metrics recorded
-    // to this (CallMetricRecorder) takes a higher precedence.
-    if (server_metric_recorder_ != nullptr) {
-      server_metric_recorder_->GetMetrics(&data);
-    }
+    // to CallMetricRecorder takes a higher precedence.
+    grpc_core::BackendMetricData data =
```

Comment:
Suggest writing this as:

```
grpc_core::BackendMetricData data;
if (server_metric_recorder_ != nullptr) {
  data = server_metric_recorder_->GetMetrics();
}
```

Reply:
I can update but wanted to check first - I did this to save one allocation if server_metric_recorder is not null, which I expect will be the case most of the time. Let me know.

Reply:
There is no dynamic allocation here; all objects are allocated on the stack.

The code you currently have is actually slightly less optimal in the case where `server_metric_recorder_` is null, since it instantiates two objects and uses a move ctor, whereas the alternative code I provided instantiates only one object.

Reply:
Updated, I missed that copy elision does not happen with a ternary operator here.